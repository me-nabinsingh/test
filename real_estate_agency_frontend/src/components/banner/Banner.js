import React, { Component } from "react";
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default class   Banner extends Component {
    render() {
        const { banner } = this.props;
        if(!banner) {
            return null;
        }
        return ( 
            <div className="position-relative overflow-hidden p-3 p-md-5 m-md-3 text-center">
                <div className="col-md-6 p-lg-5 mx-auto my-5">
                    <h1 className="display-4 font-weight-normal">{banner.title}</h1>
                    <p className="lead font-weight-normal">{banner.intro}</p>
                    <Link className="btn btn-outline-secondary" to="/planets">View all planets</Link>
                </div>
            </div>
         );
    }
}
 
Banner.defaultProps = {
    banner: {}
}

Banner.propTypes = {
    banner: PropTypes.object
}