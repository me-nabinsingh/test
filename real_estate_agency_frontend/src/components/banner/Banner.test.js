import React from 'react';
import Banner from './Banner';
import {Router } from "react-router-dom";
import renderer from 'react-test-renderer';
import history from "../../util/history";


const banner = {
    "_id":"5c0c6138dfd55477c9ca7f29",
    "title":"Real Estate Agency ",
    "intro":"Hydrogen atoms radio telescope kindling the energy hidden in matter stirred by starlight dispassionate extraterrestrial observer inconspicuous motes of rock and gas. The carbon in our apple pies globular star cluster corpus callosum concept of the number one citizens of distant epochs colonies.",
    "created_at":"2018-12-09T00:26:32.527Z",
    "updated_at":"2018-12-09T00:26:32.527Z"
};

describe('<Banner />', () => {
    it("doesn't break if banner is not passed", () => {
        const component = renderer.create(
            <Router history={history}>
                <Banner />
            </Router>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })


    it('renders without crashing', () => {
        
        const component = renderer.create(
            <Router history={history}>
                <Banner banner={banner}/>
            </Router>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});