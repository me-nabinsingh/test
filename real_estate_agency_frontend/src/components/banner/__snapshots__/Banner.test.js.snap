// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> doesn't break if banner is not passed 1`] = `
<div
  className="position-relative overflow-hidden p-3 p-md-5 m-md-3 text-center"
>
  <div
    className="col-md-6 p-lg-5 mx-auto my-5"
  >
    <h1
      className="display-4 font-weight-normal"
    />
    <p
      className="lead font-weight-normal"
    />
    <a
      className="btn btn-outline-secondary"
      href="/planets"
      onClick={[Function]}
    >
      View all planets
    </a>
  </div>
</div>
`;

exports[`<Banner /> renders without crashing 1`] = `
<div
  className="position-relative overflow-hidden p-3 p-md-5 m-md-3 text-center"
>
  <div
    className="col-md-6 p-lg-5 mx-auto my-5"
  >
    <h1
      className="display-4 font-weight-normal"
    >
      Real Estate Agency 
    </h1>
    <p
      className="lead font-weight-normal"
    >
      Hydrogen atoms radio telescope kindling the energy hidden in matter stirred by starlight dispassionate extraterrestrial observer inconspicuous motes of rock and gas. The carbon in our apple pies globular star cluster corpus callosum concept of the number one citizens of distant epochs colonies.
    </p>
    <a
      className="btn btn-outline-secondary"
      href="/planets"
      onClick={[Function]}
    >
      View all planets
    </a>
  </div>
</div>
`;
